docs/model/datamodel.md

```diff
@@ -0,0 +1,342 @@
+| `update` | Optionally specifies whether a `update` mutation is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(update: true)` |
+| `subCreate` | Optionally specifies whether a `new` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subCreate: false)` |
+| `subUpdate` | Optionally specifies whether an `updated` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subUpdate: false)` |
+| `subDelete` | Optionally specifies whether a `deleted` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subDelete: false)` |
+
+### @oneToMany
+
+`@oneToMany` is used to define a One To Many relationship.
+
+```graphql
+"""model"""
+type User {
+  id: ID!
+  """@oneToMany(field: 'author')"""
+  posts: [Post]!
+}
+
+"""@model"""
+type Post {
+  id:ID!
+  text: String!
+}
+```
+
+This example generates an `author` field on the `Post` type with a `User` resolver as well as a `userId` field internally which can be used to query across relationships.
+
+```graphql
+"""@model"""
+type Post {
+  id: ID!
+  text: String!
+  author: User
+}
+```
+
+#### Arguments
+
+| Argument | Description | Example |
+|-|-|-|
+|`field`| Specifies the name of resolver field on the foreign object. **Required**. Accepts a string value | `@oneToMany(field: 'user'`)|
+|`key`| Optionally specifies the name of foreign key field on the foreign object. Accepts a string value. Defaults to `<typeName>Id` | `@oneToMany(field: 'user', key: 'user_key')`|
+
+### @oneToOne
+
+`@oneToOne` is used to define a One To One relationship.
+
+```graphql
+type Character {
+  id:ID!
+  name: String!
+  """@oneToOne"""
+  cast_actor: Actor!
+}
+
+type Actor { ... }
+```
+
+This example generates an `character` field on the `Actor` type with a `Character` resolver as well as a `characterId` field internally which can be used to query across relationships.
+
+#### Arguments
+
+| Argument | Description | Example |
+|-|-|-|
+|`field`| Specifies the name of resolver field on the foreign object. **Required**. Accepts a string value | `@oneToMany(field: 'user'`)|
+|`key`| Optionally specifies the name of foreign key field on the foreign object. Accepts a string value. Defaults to `<typeName>Id` | `@oneToMany(field: 'user', key: 'user_key')`|
+
+### @index
+
+The `@index` annotation can be used to create an index on a specific field or a set of fields at runtime. The supported arguments are different between the different databases. See the variations for [PostgreSQL](#index-1) and [MongoDB](#index-2).
+
+> If you have relationships in your models, the relevant foreign keys are automatically indexed by Graphback, so you do not have to index them using `@index`. Similarly, custom primary keys marked with `@id` are also automatically indexed by Graphback.
+
+## PostgreSQL
+
+### ID
```

Comment:
I put this heading under PostgreSQL because the documentation will be very different for MongoDB when https://github.com/aerogear/graphback/issues/1626 is implemented.